src/mscorlib/shared/System/IO/FileStream.Windows.cs

```diff
@@ -1735,5 +1735,25 @@ private void UnlockInternal(long position, long length)
                 throw Win32Marshal.GetExceptionForLastWin32Error();
             }
         }
+        private SafeFileHandle ValidateFileHandle(SafeFileHandle fileHandle)
+        {
+            if (fileHandle.IsInvalid)
+            {
+                // Return a meaningful exception with the full path.
+
+                // NT5 oddity - when trying to open "C:\" as a Win32FileStream,
+                // we usually get ERROR_PATH_NOT_FOUND from the OS.  We should
+                // probably be consistent w/ every other directory.
+                int errorCode = Marshal.GetLastWin32Error();
+
+                if (errorCode == Interop.Errors.ERROR_PATH_NOT_FOUND && _path.Length == PathInternal.GetRootLength(_path))
+                    errorCode = Interop.Errors.ERROR_ACCESS_DENIED;
+
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, _path);
+            }
+
+            fileHandle.IsAsync = _useAsyncIO;
```

Comment:
we're sure we should be using this useAsync here, right? I recall somewhere else where we made the mistake of using the instance variable when we should have been passing a useAsync parameter instead.